tests/other/metadce/test_metadce_hello_O0.jssize

```diff
@@ -1 +1 @@
-26698
+26717
```

Comment:
Just unoptimized size, but still, why does it increase? Shouldn't we be emitting the same code? (maybe we add more assertions instrumentation if we only add it to library functions and not runtime functions?)

Reply:
This is only a 19 byte difference.   Its likely just minor difference in the way library function get emitted.. I'm not sure its work investigating, especially since the diff will be tricky to read.

Reply:
Fair enough. The diff to `[tests/other/test_unoptimized_code_size.js.size](https://github.com/emscripten-core/emscripten/pull/17403/files#diff-5664c686ff93c70307bdef746ab85d6cbd9f50bba749e427488ad617887960eb)` is almost 4K though.

Reply:
Oh ok .. I'll take a look at that one.

Reply:
Oh, I figured it out, its because this change also means that we get all of `allocate`/`addFunction`/`removeFunction` etc as part of `LEGACY_RUNTIME` which is now the default.

Would you rather I split the `LEGACY_RUNTIME` change from the moving the string functions to make this more clear?

Reply:
I see, thanks for checking.

I think it's fine in this PR to do both.